spec/template.yml

```diff
@@ -0,0 +1,183 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: template-test
+  annotations:
+    description: "Description" 
+    iconClass: "Icon" 
+    tags: "database,nosql" 
+objects:   
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: my-pod
+    labels:
+      key-pod-label: "value-pod-label"
+  spec:
+    containers:
+    - env:
+      image: registry.access.redhat.com/jboss-decisionserver-6/decisionserver63-openshift
+      name: my-container
+      ports:
+      - containerPort: 6379
+        protocol: TCP
+      securityContext:
+        privileged: true
+        seLinuxOptions:
+          level: "s0:c123,c456"
+          user: "username"
+          role: "admin"
+          type: "default"
+- apiVersion: v1
+  kind: Route
+  metadata:
+    name: my-route
+    labels:
+      key-route-label: "value-route-label"
+  spec:
+    to:
+      kind: Service
+      name: service-name
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: my-service
+  spec:
+    selector:                  
+      docker-registry: default  
+    ports:
+    - nodePort: 0
+      port: 5000               
+      protocol: TCP
+      targetPort: 5000
+- apiVersion: v1
+  kind: "BuildConfig"
+  metadata:
+    name: "my-build-config" 
+  spec:
+    runPolicy: "Serial" 
+    triggers: 
+      -
+        type: "GitHub"
+        github:
+          secret: "secret101"
+      - type: "Generic"
+        generic:
+          secret: "secret101"
+      -
+        type: "ImageChange"
+    source: 
+      git:
+        uri: "https://github.com/openshift/ruby-hello-world"
+    strategy: 
+      sourceStrategy:
+        from:
+          kind: "ImageStreamTag"
+          name: "ruby-20-centos7:latest"
+    output: 
+      to:
+        kind: "ImageStreamTag"
+        name: "origin-ruby-sample:latest"
+    postCommit: 
+        script: "bundle exec rake test"
+- apiVersion: v1
+  kind: "Build"
```

Comment:
@cben This should cover https://github.com/ManageIQ/manageiq-providers-openshift/pull/19/files#diff-c7cbabc2b2e9df5cbda6d2ecc5afecc2R292 let me know if you want to test specific attributes/relations here.